R/arc-select.R

```diff
@@ -284,22 +274,42 @@ collect_layer <- function(x, n_max = Inf, token = Sys.getenv("ARCGIS_TOKEN"), ..
 
 # utility -----------------------------------------------------------------
 
-#' Check if x is a FeatureLayer or Table object
-#'
+#' Check if x is a FeatureLayer or Table class object
+#' @keywords internal
+obj_check_layer <- function(x,
+                            class = c("FeatureLayer", "Table"),
+                            arg = rlang::caller_arg(x),
+                            call = rlang::caller_env()) {
+  check_inherits_any(
+    x,
+    class = class,
+    arg = arg,
+    call = call
+  )
+}
+
+#' Check if x inherits any of the supplied class values and error if not
+#' @inheritParams cli::cli_vec
 #' @inheritParams rlang::inherits_any
 #' @inheritParams rlang::args_error_context
 #' @keywords internal
-obj_check_arcgis <- function(x,
-                             class = c("FeatureLayer", "Table"),
-                             arg = rlang::caller_arg(x),
-                             call = rlang::caller_env()) {
-  if (rlang::inherits_any(x, class))
+check_inherits_any <- function(x,
+                               class,
+                               style = list(
+                                 "before" = "`",
+                                 "after" = "`",
+                                 "vec-last" = " or "
+                               ),
```

Comment:
Is there a reason why style is included as an argument? Shouldn't this be fixed?

Reply:
If you ever wanted to adjust the formatting style of the class information it may be nice to have it exposed but I agree it is simpler to cut it. Let me know if you spot anything else!